### PR TITLE
vb/draco metadata

### PR DESCRIPTION
**Pull request by belom88**

### CODE REVIEW

File: modules/draco/src/draco-loader.js [hunk 1]
```diff
@@ -40,7 +40,8 @@ async function parse(arrayBuffer, options, context, loader) {
     // TODO passing in options causes CI failures...
     // @ts-ignore
     return dracoParser.parseSync(arrayBuffer, {
-      extraAttributes: (options.draco && options.draco.extraAttributes) || null
+      extraAttributes: (options.draco && options.draco.extraAttributes) || null,
```
Comment (ibgreen) [inline modules/draco/src/draco-loader.js:43]: Hmm, this should be handled by default arguments for `options.draco.extraAttributes`. I merged one fix already today.

File: modules/draco/src/lib/draco-parser.js [hunk 2]
```diff
@@ -32,6 +32,7 @@ export default class DracoParser {
     /** @type {Draco3D} */
     this.draco = draco;
     this.drawMode = 'TRIANGLE';
+    this.metadataQuerier = {};
```
Comment (ibgreen) [inline modules/draco/src/lib/draco-parser.js:35]: = `null`?

File: modules/draco/src/lib/draco-parser.js [hunk 3]
```diff
@@ -44,6 +45,7 @@ export default class DracoParser {
 
   // NOTE: caller must call `destroyGeometry` on the return value after using it
   parseSync(arrayBuffer, options = {}) {
+    this.metadataQuerier = new this.draco.MetadataQuerier();
```
Comment (ibgreen) [inline modules/draco/src/lib/draco-parser.js:48]: Can we create this object in same place as other Draco objects?

**Review by ibgreen — COMMENTED** [review]
@belom88 This looks pretty good. 

I do want to see a test cases that encodes and decodes attribute metadata successfully before we land.

File: modules/draco/src/draco-loader.js (hunk 1 above)
Comment (belom88) [reply modules/draco/src/draco-loader.js:43]: This PR is from your branch "ib/draco-metadata" into "ib/draco-metadata. What fix do you mean?

File: modules/draco/src/lib/draco-parser.js (hunk 3 above)
Comment (belom88) [reply modules/draco/src/lib/draco-parser.js:48]: We will have to pass it through all methods as argument then.

File: modules/draco/src/lib/draco-parser.js (hunk 2 above)
Comment (belom88) [reply modules/draco/src/lib/draco-parser.js:35]: I causes ts errors

File: modules/draco/src/lib/draco-builder.js [hunk 4]
```diff
@@ -68,10 +71,16 @@ export default class DracoBuilder {
...
+
+    if (options.metadata) {
+      this._addGeometryMetadata(dracoPointCloud, options.metadata);
+    }
+
     const attributes = this._getAttributesFromMesh(pointcloud);
 
     // Build a `DracoPointCloud` from the input data
-    const dracoPointCloud = this._createDracoPointCloud(attributes, options);
+    this._createDracoPointCloud(dracoPointCloud, attributes, options);
```
Comment (ibgreen) [inline modules/draco/src/lib/draco-builder.js:83]: Since_createDracoMesh and _createDracoPointCloud no longer actually creates these object, I would rename them to `_buildDracoMesh` / `_buildDracoPointCloud` or something along those lines.

Was it necessary to add the metadata first? I assume that why you made this change?

**Review by ibgreen — APPROVED** [review]

**Pull request merged**